### DISCUSSION TITLE
Medium term goals

### DISCUSSION THREAD

**Discussion by j-c-cook** [j-c-cook/ghedt#92] [Goals]
# Medium Term Goals

The over-arching goal of this repository is to enable medium to large scale ground heat exchanger design installations. This post lays out medium term goals (timeline is 6 months to 1 year) for the repository. 

## Documentation

There is currently the potential for a large financial push to install large-scale community style ground heat exchangers. There is significant discussion of the features provided by `GHEDT` in my thesis, but the dissemination of that information may be more limited than peer reviewed papers. To help `GHEDT` gain traction, the current goal is to write two journal papers. 
- [ ] A paper focused on the PLAT (a more fleshed out version of Chapter 3)
- [ ] A paper focused on the automated selection and sizing (a more fleshed out version of Chapter 4)

## Further Development

The features listed here may become issues and get implemented into the repository. For now while they are atleast a couple of months out, the ideas can be listed here. 
- [ ] The ground heat exchanger object should be enhanced to handle unequal borehole depths (say deeper boreholes around the perimeter)
- [ ] `pygfunction` should be enhanced to compute g-functions with tilted boreholes (https://github.com/MassimoCimmino/pygfunction/issues/50)
- [ ] Currently, the loads accepted by `GHEDT` must be pure ground loads. Heat pump models should be introduced so that the building heating and cooling loads can be input. 

## Integration with pygfunction

This package is closely integrated with `pygfunction`. This will require maintenance to keep up with the changing API and functionality in `pygfunction`. Additionally, further research can be done into closer integration with `pygfunction`.

**Comment by NicholasFry** [comment]
@j-c-cook you could couple this library with another one from fwitte. He already has experience coupling numerical modeling engines like OpenGeoSys to heat pumps. You can contact him directly, or through https://github.com/oemof/tespy You might also consider becoming part of the Open Energy Modeling Framework for better visibility. This would at a minimum solve Further Development issue 3. further documentation: https://www.opengeosys.org/docs/benchmarks/heat-transport-bhe/3d_3bhes_array/
The nice thing is everyone here is using CoolProp.

**Reply by j-c-cook** [reply]
Thanks for the links @NicholasFry. I read their recent [paper](https://www.theoj.org/joss-papers/joss.02178/10.21105.joss.02178.pdf). The software is definitely impressive. I'll have to consider if I could utilize the [heat pump](https://tespy.readthedocs.io/en/main/tutorials_examples.html#cop-of-a-heat-pump) modeling ability.

**Reply by fwitte** [reply]
@NicholasFry, thanks for introducing tespy here and @j-c-cook nice to meet you! If you are interested we could have a meeting some time in the future and check how to implement tespy heat pump models into your software :). If you use element, you can also reach out to me at `@fwitte:matrix.org`.
